orchagent/policerorch.cpp

```diff
@@ -123,119 +142,395 @@ void PolicerOrch::doTask(Consumer &consumer)
+                /*
+                 * return here as the same error 
+                 * would be seen for the complete iteration
+                 */
+                return task_process_status::task_invalid_entry;
+            }
 
-                attrs.push_back(attr);
+            /*Enabling storm-control on port*/
+            sai_attribute_t port_attr;
+            if (storm_type == storm_broadcast)
+            {
+                port_attr.id = SAI_PORT_ATTR_BROADCAST_STORM_CONTROL_POLICER_ID;
+            }
+            else if (storm_type == storm_unknown_unicast)
+            {
+                port_attr.id = SAI_PORT_ATTR_FLOOD_STORM_CONTROL_POLICER_ID;
+            }
+            else if (storm_type == storm_unknown_mcast)
+            {
+                port_attr.id = SAI_PORT_ATTR_MULTICAST_STORM_CONTROL_POLICER_ID;
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Unknown storm_type %s", storm_type.c_str());
+                /*continue as there can be more interfaces*/
+                continue;
             }
 
+			sai_object_id_t policer_id;
             // Create a new policer
             if (!update)
-            {
-                if (!meter_type || !mode)
-                {
-                    SWSS_LOG_ERROR("Failed to create policer %s,\
-                            missing mandatory fields", key.c_str());
-                }
-
-                sai_object_id_t policer_id;
-                sai_status_t status = sai_policer_api->create_policer(
-                    &policer_id, gSwitchId, (uint32_t)attrs.size(), attrs.data());
-                if (status != SAI_STATUS_SUCCESS)
-                {
-                    SWSS_LOG_ERROR("Failed to create policer %s, rv:%d",
-                            key.c_str(), status);
-                    it++;
-                    continue;
-                }
-
-                SWSS_LOG_NOTICE("Created policer %s", key.c_str());
-                m_syncdPolicers[key] = policer_id;
-                m_policerRefCounts[key] = 0;
-            }
+			{
+				sai_status_t status = sai_policer_api->create_policer(
+						&policer_id, gSwitchId, (uint32_t)attrs.size(), attrs.data());
+				if (status != SAI_STATUS_SUCCESS)
+				{
+					SWSS_LOG_ERROR("Failed to create policer %s, rv:%d",
+							storm_policer_name.c_str(), status);
+					/* 
+					 * Returning here since the same failure 
+					 * would be seen for the full iteration
+					 */
+					return task_process_status::task_need_retry;
+				}
+
+				SWSS_LOG_NOTICE("Created storm-control policer %s", storm_policer_name.c_str());
+                m_syncdPolicers[storm_policer_name].policerOid = policer_id;
+				m_policerRefCounts[storm_policer_name] = 0;
+			}
             // Update an existing policer
             else
             {
-                auto policer_id = m_syncdPolicers[key];
+                policer_id = m_syncdPolicers[key].policerOid;
```

Comment:
key to m_syncdPolicers map is the policer name. For named policers, the name from CONFIG_DB is used directly for indexing. 
In case of storm control policers, the name is internally derived as "_interfacename_stormtype". This is used as key to the map.

Reply:
Got your comment. Key would contain multiple interface|storm_type combinations. Hence indexing the map m_syncdPolicers using key would fail. 
Replaced "key" with "storm_policer_name" inside the for loop.